main/src/library/Graph.flix

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright 2022 Nina Andrup Pedersen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+namespace Graph {
+
+    ///
+    /// Returns the transitive closure of the graph `g`.
+    ///
+    pub def closure(g: m[(t, t)]): Set[(t, t)] & Impure with Foldable[m], Boxable[t] =
```

Comment:
Should the result type be `List[(t, t)]` ? I made it `Set[(t, t)]`, because then it is easier to test, but we could just convert to `Set` in the tests themselves.